src/pgmoneta_ext/utils.c

```diff
@@ -107,4 +107,28 @@ pgmoneta_ext_check_role(Oid roleid, const char* rolename)
    list_free(role_oids);
 
    return is_success;
+}
+
+int
+pgmoneta_ext_check_privilege(Oid roleid)
+{
+   int privileges = 0;
+
+   if (pgmoneta_ext_check_superuser(roleid))
+   {
+      return PRIVILEDGE_SUPERUSER;
```

Comment:
Hmm, can a user have multiple roles? If so you should probably not return this early, rather continue checking and build bitmask along the way, using `|=`.

Reply:
They can - we need a full bitmask